### PR TITLE
Include bundled include directories as CMake SYSTEM paths

### PR DESCRIPTION
**Pull request by masterleinad**
Alternative to #14798. In the end, it appears that we simply want to suppress all warnings from dependencies and we do that already if they are not bundled. This pull request suggests also treating bundled include directories as `CMake` `SYSTEM` include directories.
This means that we should be able to get rid of most `DEAL_II_DISABLE_EXTRA_DIAGNOSTICS` as demonstrated with `Kokkos`.

### CODE REVIEW

**Review by tamiko — APPROVED** [review]
Let's add this for the time being. I will try to incorporate it into the other CMake changes for import targets.

**Comment by tjhei** [comment]
This indeed fixes the warnings that I see. :+1:

**Comment by tjhei** [comment]
CI is complaining:
```
/home/runner/work/dealii/dealii/bundled/kokkos-3.7.00/core/src/impl/Kokkos_SharedAlloc.hpp:255:15: error: ‘std::string Kokkos::Impl::SharedAllocationRecordCommon<MemorySpace>::get_label() const [with MemorySpace = Kokkos::HostSpace; std::string = std::__cxx11::basic_string<char>]’ can be marked override [-Werror=suggest-override]
```
etc.

**Comment by masterleinad** [comment]
@tjhei This compiles now for all CI checks but https://ci.tjhei.info/job/dealii-mpi/job/PR-14803/4/console times out (which doesn't appear to be related to this pull request).

**Comment by tjhei** [comment]
> times out

Yes, you are correct. I am increasing the timeout in #14808.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
     - name: build
       run: |
         cd build
-        make -j 2
+        make VERBOSE=1 -j 2
```
Comment (tjhei) [inline .github/workflows/linux.yml:60]: Are you planning to remove these again now that everything works? I don't feel strongly about it, but it produces quite a bit of output.

File: include/deal.II/differentiation/sd/symengine_number_visitor_internal.h [hunk 2]
```diff
@@ -20,18 +20,13 @@
...
 #ifdef DEAL_II_WITH_SYMENGINE
 
-DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 // Low level
 #  include <symengine/basic.h>
 #  include <symengine/dict.h>
 #  include <symengine/symengine_exception.h>
 #  include <symengine/symengine_rcp.h>
 
 // Visitor
```
Comment (tjhei) [inline include/deal.II/differentiation/sd/symengine_number_visitor_internal.h:29]: this comment is now out of place. Maybe delete?

File: source/differentiation/sd/symengine_math.cc [hunk 3]
```diff
@@ -18,15 +18,13 @@
 
 #ifdef DEAL_II_WITH_SYMENGINE
 
-DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 // Number operations
```
Comment (tjhei) [inline source/differentiation/sd/symengine_math.cc:21]: same about this comment

**Review by tjhei — APPROVED** [review]

File: .github/workflows/linux.yml (hunk 1 above)
Comment (masterleinad) [reply .github/workflows/linux.yml]: I removed it again. I don't feel strongly either. It's easy enough to add it for debugging and it's not the point of this pull request.

File: include/deal.II/differentiation/sd/symengine_number_visitor_internal.h (hunk 2 above)
Comment (masterleinad) [reply include/deal.II/differentiation/sd/symengine_number_visitor_internal.h]: Yes.

File: source/differentiation/sd/symengine_math.cc (hunk 3 above)
Comment (masterleinad) [reply source/differentiation/sd/symengine_math.cc]: Sure.

**Review by masterleinad — COMMENTED** [review]
I addressed the comments.

File: include/deal.II/differentiation/sd/symengine_number_visitor_internal.h [hunk 4]
```diff
@@ -20,18 +20,7 @@
 
 #ifdef DEAL_II_WITH_SYMENGINE
 
-DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 // Low level
```
Comment (tjhei) [inline include/deal.II/differentiation/sd/symengine_number_visitor_internal.h:23]: This comment is also out of place now.

File: include/deal.II/differentiation/ad/sacado_number_types.h [hunk 5]
```diff
@@ -67,23 +67,21 @@ DEAL_II_NAMESPACE_CLOSE
...
 #ifdef DEAL_II_TRILINOS_WITH_SACADO
 
-DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 #  include <Sacado.hpp>
 // It appears that some versions of Trilinos do not directly or indirectly
 // include all the headers for all forward and reverse Sacado AD types.
 // So we directly include these both here as a precaution.
 // Standard forward AD classes (templated)
 #  include <Sacado_Fad_DFad.hpp>
 // Reverse AD classes (templated)
```
Comment (tjhei) [inline include/deal.II/differentiation/ad/sacado_number_types.h:76]: This comment is also out of place.

**Comment by tjhei** [comment]
I take it that @tamiko checked the CMake logic, so we are good to go. Thank you, @masterleinad .

**Comment by bangerth** [comment]
Very nice! I know that a long time ago, we had tried that and for some reason determined that it didn't work. Let's hope that it works now!

**Pull request merged**